frontend/src/scenes/experiments/ExperimentView/VariantPreviewImage.tsx

```diff
@@ -0,0 +1,117 @@
+import { IconUpload, IconX } from '@posthog/icons'
+import { LemonButton, LemonDivider, LemonFileInput, LemonModal, LemonSkeleton, lemonToast } from '@posthog/lemon-ui'
+import { useActions, useValues } from 'kea'
+import { useUploadFiles } from 'lib/hooks/useUploadFiles'
+import { useState } from 'react'
+
+import { experimentLogic } from '../experimentLogic'
+import { VariantTag } from './components'
+
+export function VariantPreviewImage({
+    variantKey,
+    rolloutPercentage,
+}: {
+    variantKey: string
+    rolloutPercentage: number
+}): JSX.Element {
+    const { experiment } = useValues(experimentLogic)
+    const { updateExperimentVariantImages } = useActions(experimentLogic)
+
+    const [mediaId, setMediaId] = useState(experiment.parameters?.variant_preview_media_ids?.[variantKey] || null)
+    const [isLoadingImage, setIsLoadingImage] = useState(true)
+    const [isModalOpen, setIsModalOpen] = useState(false)
+
+    const { setFilesToUpload, filesToUpload, uploading } = useUploadFiles({
+        onUpload: (_, __, id) => {
+            setMediaId(id)
+            if (id) {
+                const updatedVariantImages = {
+                    ...experiment.parameters?.variant_preview_media_ids,
+                    [variantKey]: id,
+                }
+                updateExperimentVariantImages(updatedVariantImages)
+            }
+        },
+        onError: (detail) => {
+            lemonToast.error(`Error uploading image: ${detail}`)
+        },
+    })
+
+    return (
+        <div className="space-y-4">
+            {!mediaId ? (
+                <LemonFileInput
+                    accept="image/*"
+                    multiple={false}
+                    onChange={setFilesToUpload}
+                    loading={uploading}
+                    value={filesToUpload}
+                    callToAction={
+                        <div className="relative">
+                            <div className="Lettermark Lettermark--xlarge Lettermark--rounded Lettermark--variant-4">
+                                <IconUpload />
+                            </div>
+                        </div>
+                    }
+                />
+            ) : (
+                <div className="relative">
+                    <>
+                        <div className="text-muted inline-flex flow-row items-center gap-1 cursor-pointer">
+                            <div onClick={() => setIsModalOpen(true)} className="cursor-pointer relative">
+                                <div className="relative flex overflow-hidden select-none size-16 rounded before:absolute before:inset-0 before:border before:rounded">
```

Comment:
<img width="470" alt="Screenshot 2024-10-07 at 12 10 15" src="https://github.com/user-attachments/assets/b676967d-00b7-4c7f-987f-517c82238a29">

This is so so small – when clearly we're not hard-pressed for space, looking at all the whitespace – that it's hard to see the point of the preview.